src/pop/blockchain/pop/vbk_block_tree.cpp

```diff
@@ -35,28 +35,30 @@ void VbkBlockTree::determineBestChain(index_t& candidate,
     return;
   }
 
-  int result = cmp_.comparePopScore(candidate, state);
+  int result = cmp_.activateBestChain(candidate, state);
   // the pop state is already set to the best of the two chains
   if (result == 0) {
     VBK_LOG_DEBUG("Pop scores are equal");
     // pop scores are equal. do PoW fork resolution
     VbkTree::determineBestChain(candidate, state);
   } else if (result < 0) {
-    VBK_LOG_DEBUG("Candidate chain won");
     // the other chain won!
-    // setState(candidate) has been already done, so only update the tip
-    this->overrideTip(candidate);
+    VBK_LOG_DEBUG("Candidate chain won");
   } else {
     // the current chain is better
     VBK_LOG_DEBUG("Active chain won");
   }
 }
 
+bool VbkBlockTree::setState(const hash_t& block, ValidationState& state) {
+  return base::setState(block, state);
```

Comment:
I think you meant to do this

```
auto* index = getBlockIndex(hash);
if (!index) return false;
return this->setState(*index, state);
```

Reply:
Please add a testcase to reveal this problem

Reply:
It will call abstract base setState which looks for blockIndex by hash and redirects to specialized version of setState. I cannot make a test because there is no such issue.

Reply:
anyway, it is not clear from the code. please update